First 3 tasks with object classes

DainiusGaidam opened the pull request.

Commit:
Finished T3

Commit:
deleted t3 from login

Commit:
fixed length to 1

Commit:
First 3 tasks with object classes

nikolaj-tolkaciov commented on cypress/integration/T9.js:
```diff
@@ -0,0 +1,87 @@
+
+
+describe('Sourcebooks testLogin', function(){
+    it('T1: TestLogin with Today date', function() {
```
There is no actual value in keeping any track to T1 in here, you have a branch to do this. but when task is finished, there is no value in knowing it was firs, second or one thousand and fifth, so I would recommend removing this one

nikolaj-tolkaciov commented on cypress/integration/T9.js:
```diff
@@ -0,0 +1,87 @@
+    })
+})
+
+describe('Sorcebooks verify all user roles', function() {
```
this could be a part of previous describe:
describe('Sourcebooks testLogin',

nikolaj-tolkaciov commented on cypress/integration/T9.js:
```diff
@@ -0,0 +1,87 @@
+
+})
+
+describe('Sourcebooks creates task', function() {
```
it would be a good idea to move this to new file, like integration/task.js

linukkis replied:
I think you forgot to remove T3 from this file :)

nikolaj-tolkaciov commented on cypress/objects/loginPage.js:
```diff
@@ -0,0 +1,60 @@
+    }
+
+    getURL() {
+        return cy.url();
```
this function is kind of redundant, it just calls another function with no parameters

nikolaj-tolkaciov commented on cypress/objects/task.js:
```diff
@@ -0,0 +1,47 @@
+    }
+
+    getTaskNameFieldForFiltering() {
+        return cy.get('[class = "field--filter"]').first().find('input');
```
you can use y.get('.field--filter') for same results but save some space

nikolaj-tolkaciov commented on cypress/objects/task.js:
```diff
@@ -0,0 +1,47 @@
+    }
+
+    getViewPortWrapper() {
+       return cy.get('[class="ag-body-viewport-wrapper"').find('.ag-body-viewport')
```
same as above

nikolaj-tolkaciov commented on cypress/objects/task.js:
```diff
@@ -0,0 +1,47 @@
+        return cy.get('[type="button"]');
+    }
+
+    getTaskName() {
```
this is a bit confusing, but maybe try adding get**Random**TaskName this would bring more clarity to the functions purpose and actual behaviour

Commit:
Fixed describe titles
Fixed getTaskNameFieldForFiltering() and  getViewPortWrapper() functions to make code more beautiful

Commit:
Deleted T3 task from T9 file
Deleted getURL function from login class

Commit:
Merge branch 'dainius_gaidamavicius/develop' into dainius_gaidamavicius/feature/T9-Best_practice

Commit:
Delete URL function

Commit:
no message

Commit:
Merge branch 'dainius_gaidamavicius/feature/T9-Best_practice' of https://github.com/nikolaj-tolkaciov/sourcery-cypress into dainius_gaidamavicius/feature/T9-Best_practice

# Conflicts:
#	cypress/integration/login.js

mantasnakt commented on cypress/integration/T3-Task.js:
```diff
@@ -0,0 +1,29 @@
+describe('Sourcebooks creates task', function() {
+      it('T-3: Admin creates new task', function () {
+        task.visitTasks();
+        login.userDropDown().click({force:true});
```
command of LoginAs wasn't added here.

mantasnakt commented on cypress/integration/T3-Task.js:
```diff
@@ -0,0 +1,29 @@
+        login.getRoleOption("Admin").click();
+        login.getSubmitButton().click();
+        task.getCreateTaskButton().click();
+        let taskName = task.getRandomTaskName();
```
Randomization of the task name could be done straight in the test. It's not really related to the POM, so moving it away is not required.

mantasnakt commented on cypress/integration/T3-Task.js:
```diff
@@ -0,0 +1,29 @@
+        task.getTaskNameForm().type(taskName);
+        task.getTaskDescriptionForm().type('TASK11');
+        task.getTaskBillDrop().next().click();
+        task.inputTaskBillOption("Yes").click();
```
A perfect way to select options from the dropdowns

mantasnakt commented on cypress/integration/T3-Task.js:
```diff
@@ -0,0 +1,29 @@
+        task.inputTaskBillOption("Yes").click();
+        task.getTaskRateForm().clear();
+        task.getTaskRateForm().type(Math.random() * 5);
+        task.getTaskSubmitButton().click();
```
Just a suggestion to move it away to the class of the common objects.

mantasnakt commented on cypress/integration/T3-Task.js:
```diff
@@ -0,0 +1,29 @@
+        cy.url().should('not.contain', 'create');
+        task.visitTasks()
+        task.getTaskNameFieldForFiltering().type(taskName);
+        task.getViewPortWrapper().should('have.length', 1);
```
The test could be better. Now we don't now if the task that we have found is really the one we created. It could be our task name+some other random symbols and test will pass.
Example:
cy.get('.ellipsis' + [title="${name}"])
Geeting a cell with its title which corresponds to task name we randomized.

mantasnakt reviewed (commented):
Great job, however, I`ve noticed the absence of LoginAs command.
